README.md

```diff
@@ -166,12 +168,17 @@ Now you'll need to do steps (2) everytime you make a changes and want to build t
 
 The Makefile has following options, that are useful for developers & users. Default action is `compose-build`.
 
+**NOTE:** When you are managing your custom images, please change `DOCKER_USERNAME` with your docker account by export environment variable or passing directly to the make command. For example, if you want to publish your image, use `make publish DOCKER_USERNAME=<your-docker-username>`
+
 - `build-base`: Build openwisp-base image. It is the base image used in other images. It takes the most time to build.
 - `compose-build`: Build the openwisp images. This option must be used in most usecases, example when a user is building custom images.
 - `publish-build`: Similar to `compose-build` except the produced can be used for manually testing before publishing official images and publishing the official images using the `publish` option.
 - `runtests`: Run testcases to ensure all the services are working.
 - `develop-runtests`: Similar to `runtests`, it runs the testcases except doesn't stop the containers after running the tests which maybe desired for debugging & analyzing failing container's logs.
 - `travis-runtests`: Similar to `runtests`, it runs the testcases except used in travis builds because it doesn't clean up after running the tests and prints all the container's logs when an error occurs in the tests.
-- `clean`: Purge everything produced during building the images.
+- `clean`: Purge everything produced during building the images (only images that have 'latest' tag will be purged).
+- `release-clean`: Similar to `clean`, except it will also purge images that have 'VERSION' tag.
```

Comment:
I do think like you just said, but thats require a changing tag and repeat the lines again. I think we can use a little shell script there. I'll see...

Reply:
Oh yeah I forgot to notice that clean will delete only images tagged with `DOCKER_USERNAME` too. That's why making `clean` to be universal need a shell script, because current implementation only purged images with `DOCKER_USERNAME` as registry and current image version as tag. The caveat is just it will also purge unintended images, for example if user have two images, their custom images (which they build with `make develop`) and openwisp latest image (that they pull the images by theirself), then user want to purge only image with their own username and they don't want to purge openwisp images, then universal `clean` can't do that job because it will delete both nevertheless. Tracking which one is comes from `make` is hard and probably will generate some unintended changes. If its safe to make clean to be universal, then I'll look forward to use shell script instead.

Reply:
In my head, I think `clean` is very aggressive, and if someone is using it, I assume they want to remove everything, I think it's safe to assume if someone is using `clean` they just want to purge all the tags, the new, the old, everything.

Also, can you please mention that clean is an agressive `clean` this to the documentation?

We can have a soft clean in the future that take cares of the tags and what not in the future as per the requirements, but this `clean` came into existence for removing everything and I think we should stay true to that goal and focus on the rest later outside this issue.

What do you think? :smile:

Reply:
I see, Okay I will try my best to not using a shell script 😊

Reply:
When I think of `make clean` I mostly think of working with C code and I agree that it is usually there to restore the local repository to its original state.  That would mean purging everything created by make since cloning the repository.